packages/loader/container-loader/src/container.ts

```diff
@@ -1493,6 +1505,17 @@ export class Container
 
 		let opsBeforeReturnP: Promise<void> | undefined;
 
+		const freezeAtSeqNumListener = (message: ISequencedDocumentMessage) => {
+			assert(loadToSequenceNumber !== undefined, "loadToSequenceNumber should be defined");
+			if (message.sequenceNumber >= loadToSequenceNumber) {
+				// Immediately pause the inbound queue and disconnect the container.
+				void this.deltaManager.inbound.pause();
+				void this.deltaManager.outbound.pause();
+				this.disconnect();
```

Comment:
What if user does not want to disconnect? why these things are coupled?

Reply:
Also how user could get back on track (if that's what user wants)? I.e. unpause queues?

Reply:
> What if user does not want to disconnect? why these things are coupled?

I suppose they do not have to be coupled, but I never really thought of why this is desired. What do you think are the reasons they may want to stay connected in this state?

> Also how user could get back on track (if that's what user wants)? I.e. unpause queues?

If we want to support this we can allow them to simply call `connect()` to get back on track. I'll validate this scenario and add testing for it shortly

Reply:
We are building a platform; we can't be sure how our product will be used. I'll ask a reverse question - do you see this solution being used anywhere else outside of one specific use case?

Reply:
Pausing is probably sufficient here without disconnecting.  Let's also use more-direct terminology then, like "pauseAfterLoad" rather than "freezeAfterLoad" to better explain the state the Container will be in using our existing concepts.

Reply:
Agree on the platform-centric approach. We probably should consider value of ever exposing concept of "ops" and "opsBeforeReturn" during load, as user needs to understand what it means for them and their use cases. The same comments for op caching option. I would suggest lets leave that as a separate effort.

@scottn12 why container needs to be disconnected? How does user get out of "paused" state?

Reply:
I agree, pausing should be sufficient. My latest commits remove the disconnect from the logic. 

@ssimic2, as per getting out of the "paused" state, we do not have a public API to do this. If the user wants to force it, they can call`container.deltaManager.inbound.resume()`/`container.deltaManager.outbound.resume()`. I don't know if we want to add functionality to officially support this behavior, since in the past we explicitly removed the ability to pause/unpause the container freely (it can be dangerous since the pause/unpause logic is a bit finicky). Additionally, we don't currently have a use case to unpause the container.